src/serde.rs

```diff
@@ -172,4 +220,20 @@ mod tests {
             ],
         );
     }
+
+    #[test]
+    fn test_buffer_alignment() {
+        const SIZE_FIRST: usize = 11;
+        const SIZE_SECOND: usize = 13;
+        let buffer = Buffer::<SIZE_FIRST, SIZE_SECOND>::new();
+
+        // Make sure that the struct allocated continuous memory, as we exploit that fact with the
+        // `as_slice` and `as_mut_slice()` methods.
+        let start_first = ptr::addr_of!(buffer.first) as *const u8;
+        let start_second = ptr::addr_of!(buffer.second) as *const u8;
+        #[allow(unsafe_code)]
+        unsafe {
+            assert_eq!(start_second.offset_from(start_first), SIZE_FIRST as isize);
+        };
```

Comment:
This is testing implementation details. Wouldn't it be better to just write to the slice via `as_mut_slice` and read from it again via `as_slice` and compare the results?

If that works without panicking, your unsafe code is correct.

Reply:
Good point, I added another test.